packages/pyolite-kernel/src/worker.ts

```diff
@@ -100,6 +101,24 @@ async function sendComm(
   });
 }
 
+/**
+ * Send a input request to the front-end.
+ *
+ * @param prompt the text to show at the prompt
+ * @param password Is the request for a password?
+ */
+async function sendInputRequest(prompt: string, password: boolean) {
+  const content = {
+    prompt,
+    password
+  };
+  postMessage({
+    type: 'input_request',
+    parentHeader: formatResult(kernel._parent_header['header']),
+    content
+  });
+}
+
```

Comment:
You are missing a function, let's call it `input`. It's the function you should set to the interpreter with `interpreter.input_request = input`.

This function might be tricky to implement, you should find a way for this to be a blocking call which resolves only when the input-reply has been received.

So:
- send input request to the `kernel.ts` with `sendInputRequest`
- wait for input reply. This may be done by waiting for a Promise that is resolved only when the `input-reply` is resolved. 
- return the content of the input reply to Python